Added locationResponseVO to vendor response info

Bozhokmaria opened the pull request:
Added locationResponseVO to vendor response info as frontend team asked

Bozhokmaria committed:
Added locationResponseVO to vendor response info

dmitriychernov7 commented on src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java:
```diff
@@ -25,7 +26,7 @@ public VendorMapper(LocationService locationService) {
 
     public VendorResponseVO toVO(Vendor vendor) {
```
please add tests to check that mapping is correct

Bozhokmaria replied:
added

ahlukhouski2 commented on src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java:
```diff
@@ -47,15 +48,5 @@ private void provideLocationDependencies(VendorRequestVO request, Vendor vendor)
-        modelMapper.addMappings(createSkipPropertyMap());
-    }
-
-    private PropertyMap<VendorRequestVO, Vendor> createSkipPropertyMap() {
```
why it is removed?

ahlukhouski2 commented on src/main/java/com/exadel/discountwebapp/vendor/vo/VendorRequestVO.java:
```diff
@@ -3,9 +3,13 @@
 @SuperBuilder
 @NoArgsConstructor
 public class VendorRequestVO extends VendorBaseVO {
+    private Long locationId;
```
@NotNull? (no need to define @Positive)

Bozhokmaria replied:
+

ahlukhouski2 commented on src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java:
```diff
@@ -25,7 +26,7 @@ public VendorMapper(LocationService locationService) {
     public VendorResponseVO toVO(Vendor vendor) {
         VendorResponseVO response = modelMapper.map(vendor, VendorResponseVO.class);
-        response.setLocationId(vendor.getLocation().getId());
+        response.setLocationResponseVO(modelMapper.map(vendor.getLocation(), LocationResponseVO.class));
```
probably LocationMapper can be injected and reused here to map Location -> LocationResponseVO

Bozhokmaria replied:
+

ahlukhouski2 commented on src/main/java/com/exadel/discountwebapp/vendor/vo/VendorResponseVO.java:
```diff
@@ -10,4 +11,6 @@
     private Long id;
-}
+
+    private LocationResponseVO locationResponseVO;
```
let's name this field simply 'location'

Bozhokmaria replied:
+

Bozhokmaria committed:
Merge branch 'develop' into add-location-to-vendor-responce

Bozhokmaria committed:
edited according to comments
+ added test for VendorMapper
+ edited integration test because of changes in Vendor VOs
